move all DSOs to large pages in LD_PRELOAD lib and also fix a few issues

rdementi opened the pull request.

suresh-srinivas commented on large_page-c/large_page.c:
```diff
@@ -287,11 +287,11 @@ static void AlignRegionToPageBoundary(mem_range* r) {
   }
 
-  if (r->to - r->from < HPS) {
+  if (r->to - r->from < HPS || r->from > r->to) {
```
Does this condition happen where from > to? We were treating it as an invalid region

rdementi replied:
"r->from > r->to" condition occurs for smaller regions after AlignRegionToPageBoundary is called ('from' is aligned up and 'to' is aligned down). This part of commit is not critical. I can drop it if needed.

rdementi replied:
before AlignRegionToPageBoundary : from < to

suresh-srinivas replied:
Thanks, that makes sense.

suresh-srinivas reviewed (approved):
Thanks for the contribution. This Looks good to me. Is it possible to put the printfs under debug and use stderr? I don't know how it would work when used under systemd

uttampawar commented on large_page-c/lp_preload.c:
```diff
@@ -1,7 +1,49 @@
 #include "large_page.h"
 
+void printErr (map_status status, const char * lib)
+{
```
nipick. Please put open curly braces on the same line as function definition matching with rest of the coding standard.

uttampawar commented on large_page-c/lp_preload.c:
```diff
@@ -10,7 +52,23 @@ void __attribute__((constructor)) map_to_large_pages() {
+  regex_t ignoreReg;
+  const char * ignoreStr = secure_getenv("LP_IGNORE");;
+  if (ignoreStr == NULL || regcomp(&ignoreReg, ignoreStr, REG_EXTENDED) != 0)
+  {
```
nitpick. Same as before about the curly braces.

uttampawar reviewed (approved):
LGTM.

rdementi commented:
@uttampawar @suresh-srinivas  Thanks a lot for your comments. I think I have addressed your comments in the latest commit I have pushed. Please review

uttampawar reviewed (approved):
LGTM
